contracts/base/escrow/dapps/YVault.sol

```diff
@@ -0,0 +1,144 @@
+    @author develop@teller.finance
+ */
+contract YVault is YVaultInterface, BaseEscrowDapp {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    
+
+    /** External Functions */
+    /**
+        @notice Deposits the specified amount of the native unwrapped token (same as token() returns) into the Vault
+        @param tokenAddress The address of the token being deposited
+        @param amount The amount of tokens to be deposited into the vault
+     */
+    function deposit(
+        address tokenAddress,
+        uint256 amount
+    )
+        public
+        onlyOwner()
+    {
+        IVault iVault = _getYVault(tokenAddress);
+        IERC20 underlyingToken = IERC20(iVault.underlying());
+        uint256 tokenBalanceBeforeDeposit = underlyingToken.balanceOf(address(this));
+        IERC20(tokenAddress).safeApprove(address(iVault), amount);
+        uint256 result = iVault.deposit(amount);
+        require(result == 0, "YEARN_DEPOSIT_ERROR");
+        uint256 tokenBalanceAfterDeposit = underlyingToken.balanceOf(address(this));
+        require(
+            tokenBalanceAfterDeposit > tokenBalanceBeforeDeposit,
+            "YEARN_BALANCE_NOT_INCREASED"
+        );
+        _tokenUpdated(tokenAddress);
+
+        emit YearnDeposited(
+            tokenAddress,
+            address(iVault),
+            amount,
+            tokenBalanceAfterDeposit
+        );
+    }
+
+    /**
+        @notice Withdraws the specified amount of the native unwrapped token (same as token() returns) from the Vault
+        @param tokenAddress The address of the token being deposited
+        @param amount The amount of tokens to be deposited into the vault
+     */
+    function withdraw(
+        address tokenAddress,
+        uint256 amount
+    ) 
+        public
+        onlyOwner()
+    {
+        IVault iVault = _getYVault(tokenAddress);
+        _withdraw(iVault, amount);
+    }
+
+    /**
+        @notice Returns the price of the Vault's wrapped token, denominated in the unwrapped native token
+        @notice Calculation is: nativeTokenBalance/yTokenTotalSupply, 
+            - nativeTokenBalance is the current balance of the native token (example DAI) in the Vault
+            - yTokenTotalSupply is the total supply of the Vault's wrapped token (example yDAI)
+        @param tokenAddress The address of the underlying token for the associated yVault
+        @return The token price
+     */
+    function getPricePerFullShare(address tokenAddress) 
+        public
+        view
+        returns (uint256)
+    {
+        IVault iVault = _getYVault(tokenAddress);
+        return iVault.getPricePerFullShare();
+    }
+
+    /** Internal Functions */
+    /**
+        @notice Redeems funds from a yVault from a previous deposit
+        @param iVault The instance of the yVault
+        @param amount The amount of funds to withdraw from the vault
+     */
+    function _withdraw(
```

Comment:
Don't really need this internal function since it's only used once

Reply:
Was thinking about adding a withdrawAll() external method that would call this internally